oxidation/crates/parsec_api_types/src/manifest.rs

```diff
@@ -243,66 +260,36 @@ fn generate_local_author_legacy_placeholder() -> DeviceID {
     LEGACY_PLACEHOLDER.clone()
 }
 
-macro_rules! impl_dump_sign_and_encrypt {
-    ($name:ident) => {
-        impl $name {
-            pub fn dump_sign_and_encrypt(
-                &self,
-                author_signkey: &::parsec_api_crypto::SigningKey,
-                key: &::parsec_api_crypto::SecretKey,
-            ) -> Vec<u8> {
-                let serialized = rmp_serde::to_vec_named(&self).unwrap_or_else(|_| unreachable!());
-                let mut e =
-                    ::flate2::write::ZlibEncoder::new(Vec::new(), flate2::Compression::default());
-                use std::io::Write;
-                e.write_all(&serialized).unwrap_or_else(|_| unreachable!());
-                let compressed = e.finish().unwrap_or_else(|_| unreachable!());
-                let signed = author_signkey.sign(&compressed);
-                key.encrypt(&signed)
-            }
+/*
+ * Blocksize
+ */
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Blocksize(u64);
```

Comment:
don't we want a `impl Deref for Blocksize` to have transparent access to the u64 ?